Make alert accessible

felippenardi opened the pull request:
#### What is the purpose of this pull request?
This is a non-breaking change to add some accessibility features to the alert.

<!--- Describe your changes in detail. -->

#### What problem is this solving?
[Running workspace](https://accessibility--content.myvtex.com/admin/app/new-cms/edit/new)

1- Allow developers to use `focusOnOpen` property to make the cursor focus change to either the action or dismiss buttons
2- Make screen readers announce the alert error as soon as it pops up in the screen


<!--- What is the motivation and context for this change? -->

#### How should this be manually tested?
You can go to the workspace link above and try saving a document without filling any information. This will trigger an error. Then you should try repeating the action and, in the second time, dismissing the alert.

You should be able to do all this using just your keyboard.

If you have voice over activated, you should be able to hear the entire experience seamlessly.

#### Screenshots or example usage
![alert](https://user-images.githubusercontent.com/1354492/82239671-8699b180-990f-11ea-9fbf-04e608e1d285.gif)


#### Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

felippenardi commented:
I'll add the documentation and changelog in a bit

emersonlaurentino commented:
@felippenardi great job, we need more a11y components. 

Can you add changelog?

Klynger commented on react/components/Alert/index.js:
```diff
@@ -109,11 +137,17 @@ Alert.propTypes = {
+  /** Set focus to the first focusable element inside alert, which should be the "action" when available or the "close" button */
+  focusOnOpen: PropTypes.bool,
+  /** Define element to be focused when the alert is dimissed */
+  focusAfterClosed: PropTypes.shape({ current: PropTypes.instanceOf(Element) }),
```
I think this prop is not needed, you can get `document.activeElement` and hold so when the user dismisses the alert its focuses on this element. Another thing to think about is when the user dismiss the alert, but the focus wasn't in the alert anymore, so you shouldn't mess with the current focused element

felippenardi replied:
Just talked to Klynger on discord, and he suggested the idea of replacing both these properties for a single property that receives an ENUM.

The enum would have three options:
- Focus on action or dismiss when the alert opens
- Do the above and also register the document.activeElement and return to it when dismiss is triggered

felippenardi replied:
After analyzing the behavior I noticed a few scenarios where changing the focus after closing may be problematic, so I've decided not to implement it.

felippenardi commented:
@emersonlaurentino @Klynger @iaronaraujo I've removed the focus feature for when the alert is dismissed because it would potentially cause accessibility problems.

I've also added changelog, let me know if we are good to merge.
